src/pages/login/login-container.tsx

```diff
@@ -28,12 +30,29 @@ export class LoginContainer extends React.Component<LoginContainerProperties> {
   static mapActions(_props: LoginContainerProperties): Partial<LoginContainerProperties> {
     return {
       switchLoginStage,
+      setEmailSubmitted,
     };
   }
 
   handleSelectionChange = (selectedOption: string) => {
     const { switchLoginStage } = this.props;
-    switchLoginStage(selectedOption === 'web3' ? LoginStage.Web3Login : LoginStage.EmailLogin);
+    switch (selectedOption) {
+      case 'web3':
+        switchLoginStage(LoginStage.Web3Login);
+        break;
+      case 'email':
+        switchLoginStage(LoginStage.EmailLogin);
+        break;
+      case 'reset':
+        switchLoginStage(LoginStage.ResetPassword);
+        break;
+      case 'backToLogin':
+        switchLoginStage(LoginStage.EmailLogin);
+        this.props.setEmailSubmitted(false);
```

Comment:
I don't follow this part. Could you please explain?

Reply:
Yeah sure! 

In `src/pages/login/login-component.tsx`, line:

```
<button {...cn('button-link')} onClick={() => this.props.handleSelectionChange('backToLogin')}>
```

If the button is clicked, we set the `selectedOption` to `backToLogin`. So in the switch statement above, if the case is equal to `backToLogin` then two things happen:
1. The login stage is set to `EmailLogin` this means the UI will display the log in with email screen. 
2. The state for 'has an email been submitted' is set back to its initial state `false`. This means that if the user navigates back to the reset screen, they will not see the success message, but instead the form to enter their email.

Let me know if that's not clear !